Python/read_feh.py

```diff
@@ -125,14 +132,27 @@ def read_header_file(filename:str, sample:str):
 
     # Open header file
     with open(filename, 'rb') as file:
-
-        # Read year from the first 10 bytes
-        if sample == "input": 
-            year = file.read(4)
-            year = int.from_bytes(year, byteorder="little", signed=True)
-        elif sample == "output": 
-            year = file.read(10)
-            year = int(year)
+        # Get the first 10 bytes
+        first_10_bytes = file.read(10)
+
+        # Check if the first 4 bytes of the first 10 represent the year 2006
+        if int.from_bytes(first_10_bytes[:4], byteorder="little", signed=True) == 2006:
+            sample = 'input'
+            year = int.from_bytes(first_10_bytes[:4], byteorder="little", signed=True)
+            # Go back to the 4th byte to ensure the file is in the correct place
+            file.seek(4)
+        
+        # Check if the first 10 bytes represent the year 2060
+        elif int(first_10_bytes) == 2060:
```

Comment:
This has the same problem as above (i.e., using a value that's likely to change) but also it is not accurate with respect to the encoding of these 10 bytes. In the output file, first 10 bytes contain ASCII codes. First 6 bytes are the code for white space, 0x20, and the last 4 bytes are ASCII codes for the last simulation year, which is 2060 in the sample you're working with, but is 2100 in most other samples. See the R implementation.

Reply:
Addressed